### PR TITLE
Added parallel OS package publishing and rolling release channel

### PR DESCRIPTION
**Pull request by fheinecke**
This adds parallel publishing for OS packages published by Drone, as well as a new rolling release channel.

This requires https://github.com/gravitational/teleport.e/pull/1766 to be merged first, and an `e` ref update.

Note that by line count about half the changes are to the generated .drone.yml.

### CODE REVIEW

File: dronegen/gha.go
Comment (r0mant) [inline dronegen/gha.go:88]: Why do we need a sleep here and why do sleep intervals linearly increase by 10 seconds for each subsequent workflow?

**Review by r0mant — APPROVED** [review]
lgtm once you've added a comment clarifying why the sleep delay is necessary

**Comment by camscale** [comment]
@fheinecke Is this still "do-not-merge"? I tend to hold of reviewing such PRs as I'm not sure how much change is still going to happen before merging. Was it just waiting on the merge of the related PR and an `e` ref update? Happy to review this now if its ready.

**Comment by fheinecke** [comment]
@camscale yes it is. This is "do not merge" because I need to synchronize the deployment of this PR, the teleport.e PR, and an immediate follow-up migration. In short, I need both PRs ready for me to hit the "merge" button, along with a maintenance window, before I can remove this label.

The PR is ready to review and is not expected to have significant changes unless they are required as a result of reviews.

**Review by camscale — COMMENTED** [review]
I'm sorry but I think I am missing something here. I don't understand what the purpose of the sleeps are or why the sleep duration keeps increasing. I also don't see where parallelism is added. I see dependencies being added but what makes things run in parallel.

So as I write this, I think maybe I understand - does drone automatically serialise steps unless there is a `depends_on`? If so I think that makes a bit more sense now - adding `depends_on` specifies a non-serial graph. All the sleep steps wait on the last setup step and the parallel work happens by each work step depending on its sleep step. Presumably the wait steps are there so you don't fire all the workflows simultaneously (give the autoscaler a change to grow the runner pool?)

I also assume that if you add any `depends_on`, the whole pipeline becomes parallel, so any steps without a `depends_on` will trigger at the same time at the start of the pipeline? And this explains why you need to add dependencies in the setup steps? Otherwise they'll start before the checkout code step completes?

Anyway, if I figured that out correctly it was from looking at the changes to `.drone.yml` as I didn't understand what the code was actually doing. I think the code really needs some comments to explain this - if I landed on this code I think I would remain mystified. I have come back to this PR 5 times now to read the code to try to get it and now I decided to look at `.drone.yml` could I piece it together.

Let me know if I got it right and I'll approve assuming comments are coming. Otherwise, I'm afraid I don't know what is going on. Sorry.

**Review by tcsc — COMMENTED** [review]
I think I'm on the same page as @camscale - although I had to dive into the [Drone docs](https://docs.drone.io/pipeline/docker/syntax/parallelism/) to get there.

Can you please add some commentary the appropriate parts of `dronegen` to help those who come after? I know I started off on the wrong foot because I am deeply suspicious of anything to do with parallelism that adds seemingly-random sleeps to things. 

An ascii-art diagram of the resulting DAG would be super useful.

**Comment by fheinecke** [comment]
@camscale That's exactly right.

@camscale / @tcsc I added a comment regarding the sleep time along with a link to the relevant documentation.

With respect to additional comments and documentation, where/how/what do you guys suggest I add? I agree that dronegen-generated parallelism is very confusing and difficult to follow, however I'm not really sure what I could feasibly add to make this easier. I've hit this issue several times on other projects (such as container building which is also parallelized) and haven't been able to come up with a decent answer.

Regarding an ASCII drawing, I think think this is a great idea conceptually however I think it's pretty hard to maintain longer term. Any change to the code then requires an update to the drawing, and if they are out of sync then the drawing is at best confusing and misinformative at worst. I can do this if everybody thinks this is the best approach - I'm just not sure that it is.

As an aside, Drone v2 has much better support for parallelism and will even generate DAG images for us. I very much don't think that this adds enough value to switch our (now deprecated) Drone v1 install, but if this is a big enough issue then I might be able to write a tool that runs Drone v2 in a container and uses it to generates a DAG image of the pipeline.

**Comment by fheinecke** [comment]
One unrelated thing - it looks like dronegen did something weird and reordered several completely unrelated pipelines. Does anybody have any idea why this might have happened? I don't recall seeing this before.

**Review by camscale — APPROVED** [review]

**Comment by camscale** [comment]
> One unrelated thing - it looks like dronegen did something weird and reordered several completely unrelated pipelines

I'm not seeing this. Perhaps it reordered back again?

**Comment by fheinecke** [comment]
> Perhaps it reordered back again?

Definitely not - see https://github.com/gravitational/teleport/pull/28659/files#diff-b54b39f1afced2465e1f3641db9d5bbf4f3a7fcf890996dfedd3c197bcb7f8c7R4294 for an example

I don't _think_ there are any consequential changes, but with 1340 diffs it's a little hard for me to say for certain...

**Pull request merged**
